data-prepper-plugin-framework/src/test/java/org/opensearch/dataprepper/plugin/PluginBeanFactoryProviderTest.java

```diff
@@ -112,8 +113,32 @@ void testCreatePluginSpecificContext() {
     void testCreatePluginSpecificContext_with_empty_array() {
         when(context.getParent()).thenReturn(context);
         final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
-        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{});
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{}, null);
         assertThat(beanFactory, notNullValue());
         assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestComponent.class));
     }
+
+    @Test
+    void testCreatePluginSpecificContext_with_empty_array_with_plugin_config() {
+        when(context.getParent()).thenReturn(context);
+        final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
+        TestPluginConfiguration config = new TestPluginConfiguration();
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{}, config);
+        assertThat(beanFactory, notNullValue());
+        assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestComponent.class));
+        assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestPluginConfiguration.class));
+    }
+
+    @Test
+    void testCreatePluginSpecificContext_with_plugin_config() {
+        when(context.getParent()).thenReturn(context);
+        final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
+        TestPluginConfiguration config = new TestPluginConfiguration();
+        config.setRequiredString("required_string_value");
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{TestComponent.class}, config);
+        assertThat(beanFactory, notNullValue());
+        assertThat(beanFactory.getBean(TestComponent.class), notNullValue());
+        assertThat(beanFactory.getBean(TestPluginConfiguration.class), notNullValue());
+        assertEquals("required_string_value", beanFactory.getBean(TestPluginConfiguration.class).getRequiredString());
+    }
```

Comment:
Please also add a unit test for this condition: `!(configuration instanceof PluginSetting)`

Reply:
Added additional test case to cover this scenario 👍